### PR TITLE
Improvements to command routing

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR:
  * Introduces two new types `DispatchedCommand` and `DispatchedEvent`
  * Alters `CommandRouted` to use `DispatchedCommand`.
  * Migrates to new version of Testutil.

The Spine version was advanced to `0.10.50-SNAPHOT`.

### CODE REVIEW

File: client/src/main/java/io/spine/client/CommandFactory.java [hunk 1]
```diff
@@ -216,10 +216,10 @@ static CommandContext createContext(@Nullable TenantId tenantId,
                                         UserId userId,
                                         ZoneOffset zoneOffset,
                                         int targetVersion) {
-        CommandContext.Builder builder = newContextBuilder(tenantId, userId, zoneOffset);
-        CommandContext result = builder.setTargetVersion(targetVersion)
-                                             .build();
-        return result;
+        CommandContext.Builder result =
+                newContextBuilder(tenantId, userId, zoneOffset)
+                        .setTargetVersion(targetVersion);
```
Comment (armiol) [inline client/src/main/java/io/spine/client/CommandFactory.java]: This portion of expression fits the previous line.

File: core/src/main/proto/spine/core/command.proto [hunk 2]
```diff
@@ -180,3 +179,13 @@ enum CommandStatus {
     // A command was rejected during processing. No events were generated.
     REJECTED = 5;
 }
+
+// Data from a `Command` which is accessible from the code to which the command was dispatched.
```
Comment (armiol) [inline core/src/main/proto/spine/core/command.proto]: We need to mention that we mean consumer codebase, not the Spine's internal `...Endpoint` code or `...Repository` code. E.g.

```
 Data from a `Command` which was accessible from the command handler method to which it was dispatched.
```

File: core/src/main/proto/spine/core/command.proto [hunk 3]
```diff
@@ -180,3 +179,13 @@ enum CommandStatus {
...
+message DispatchedCommand {
+
+    // The message of the command.
+    google.protobuf.Any message = 2 [(required) = true];
```
Comment (armiol) [inline core/src/main/proto/spine/core/command.proto]: We need to start indexing from `1`.

File: core/src/main/proto/spine/core/event.proto [hunk 4]
```diff
@@ -136,3 +135,13 @@ enum EventValidationError {
     // The event message is invalid according to custom Protobuf options set.
     INVALID_EVENT = 2;
 }
+
+// Data from an `Event` which is accessible from the code to which the event was dispatched.
```
Comment (armiol) [inline core/src/main/proto/spine/core/event.proto]: Similar to the comment above,
```
Data from an `Event` which was accessible from the event handler method (e.g. applier, reactor or subscriber) to which it was dispatched.
```

File: server/src/main/java/io/spine/server/procman/ProcessManager.java [hunk 5]
```diff
@@ -104,8 +104,11 @@ void setCommandBus(CommandBus commandBus) {
         this.commandBus = checkNotNull(commandBus);
     }
 
-    /** Returns the {@code CommandBus} to which post commands produced by this process manager. */
+    /**
+     * Returns the {@code CommandBus} to which post commands produced by this process manager.
```
Comment (armiol) [inline server/src/main/java/io/spine/server/procman/ProcessManager.java]: I think it better reads as 
```
Returns the {@code CommandBus} to which the commands produced by this process manager are to be posted.
```

File: server/src/test/java/io/spine/core/EventsTest.java [hunk 6]
```diff
@@ -131,9 +131,9 @@ void actor() {
         @Test
         @DisplayName("producer")
         void producer() {
-            final StringValue msg = unpack(context.getProducerId());
+            StringValue msg = unpack(context.getProducerId());
 
-            final String id = getProducer(context);
+            String id = (String)getProducer(context);
```
Comment (armiol) [inline server/src/test/java/io/spine/core/EventsTest.java]: A space symbol is missing. Please reformat.

File: server/src/test/java/io/spine/server/procman/AbstractCommandRouterTest.java [hunk 7]
```diff
@@ -117,33 +90,15 @@ public void onError(CommandEnvelope envelope, RuntimeException exception) {
...
+        // The command message we route.
+        Message sourceMessage = toMessage(getClass().getSimpleName());
+        // The context of the command that we route.
+        CommandContext sourceContext = requestFactory.createCommandContext();
 
-    static List<StringValue> unpackAll(List<Any> anyList) {
-        return anyList.stream()
-                      .map(AnyPacker::<StringValue>unpack)
-                      .collect(Collectors.toList());
+        // The object we test. */
```
Comment (armiol) [inline server/src/test/java/io/spine/server/procman/AbstractCommandRouterTest.java]: `*/` portion seems a redundant leftover.

**Comment by alexander-yevsyukov** [comment]
@armiol, PTAL.

**Pull request merged**